datafusion/common/src/scalar.rs

```diff
@@ -2156,30 +2157,58 @@ impl ScalarValue {
     ///
     /// assert_eq!(scalar_vec, expected);
     /// ```
-    pub fn convert_array_to_scalar_vec(array: &dyn Array) -> Result<Vec<Vec<Self>>> {
-        let mut scalars = Vec::with_capacity(array.len());
+    pub fn convert_list_array_to_scalar_vec(array: &dyn Array) -> Result<Vec<Vec<Self>>> {
+        let mut scalars_vec = Vec::with_capacity(array.len());
+
+        if as_list_array(array).is_ok() {
+            let list_arr = as_list_array(array)?;
+            for index in 0..list_arr.len() {
+                let scalars = match list_arr.is_null(index) {
+                    true => Vec::new(),
+                    false => {
+                        let nested_array = list_arr.value(index);
+                        ScalarValue::convert_list_array_to_scalar_vec(&nested_array)?
+                            .into_iter()
+                            .flatten()
+                            .collect()
+                    }
+                };
+                scalars_vec.push(scalars);
+            }
+        } else {
+            let scalars = ScalarValue::convert_non_list_array_to_scalars(array)?;
```

Comment:
This seems like it would result in some very confusing behavior -- a single row vec -- If I pass a non `ListArray` to a function called `convert_list_array_to_scalar_vec` I would prefer to get an error back so I can track down the bug in my code rather than get back something unexpected and have to know to check for that